modules/gateway/peers.go

```diff
@@ -52,8 +53,23 @@ func (g *Gateway) addPeer(p *peer) {
 	go g.listenPeer(p)
 }
 
+// randomPeer returns a random peer from the gateway's peer list.
+func (g *Gateway) randomPeer() (modules.NetAddress, error) {
+	if len(g.peers) > 0 {
+		r := rand.Intn(len(g.peers))
```

Comment:
the problem is that peer selection does play into things like isolation attacks. Though it seems harmless here, there's always a chance that someone figures out how to abuse non-cryptographic random numbers to manipulate the network for their purposes (i.e. target specific peers for DOS, or something to that effect. If they can guess what your selection will be, it's possible that they can use it against you.)